src/party_menu_items.c

```diff
@@ -0,0 +1,1015 @@
+    UseItemOnMonInParty(partyMenu->args->party, partyMenu->args->itemId, partyMenu->partyMonIndex, 0, PartyMenu_GetCurrentMapSec(partyMenu), HEAP_ID_PARTY_MENU);
+    Pokemon *mon = Party_GetMonByIndex(partyMenu->args->party, partyMenu->partyMonIndex);
+    int hp = GetMonData(mon, MON_DATA_HP, NULL);
+    String *string;
+    if (partyMenu->monsDrawState[partyMenu->partyMonIndex].hp == 0) {
+        string = NewString_ReadMsgData(partyMenu->msgData, msg_0300_00071);
+    } else if (partyMenu->monsDrawState[partyMenu->partyMonIndex].hp == hp) {
+        string = NewString_ReadMsgData(partyMenu->msgData, msg_0300_00072);
+    } else {
+        string = NewString_ReadMsgData(partyMenu->msgData, msg_0300_00065);
+    }
+    BufferBoxMonNickname(partyMenu->msgFormat, 0, Mon_GetBoxMon(mon));BufferIntegerAsString(partyMenu->msgFormat, 1, hp - partyMenu->monsDrawState[partyMenu->partyMonIndex].hp, 3, PRINTING_MODE_LEFT_ALIGN, TRUE);
+    StringExpandPlaceholders(partyMenu->msgFormat, partyMenu->formattedStrBuf, string);
+    String_Delete(string);
+    int statusIcon = Pokemon_GetStatusIconId(mon);
+    PartyMenu_DrawMonStatusIcon(partyMenu, partyMenu->partyMonIndex, statusIcon);
+    if (statusIcon == PARTY_MON_STATUS_ICON_OK) {
+        partyMenu->monsDrawState[partyMenu->partyMonIndex].status = PARTY_MON_STATUS_ICON_OK;
+        PartyMenu_PrintMonLevelOnWindow(partyMenu, partyMenu->partyMonIndex);
+    }
+    sub_0207A7F4(partyMenu, partyMenu->partyMonIndex);
+    partyMenu->itemUseCallback = PartyMenu_ItemUseFunc_HPRestoreAnimLoop;
+    PlaySE(SEQ_SE_DP_KAIFUKU);
+    return PARTY_MENU_STATE_ITEM_USE_CB;
+}
+
+static int PartyMenu_ItemUseFunc_HPRestoreAnimLoop(PartyMenuStruct *partyMenu) {
+    int hp = GetMonData(Party_GetMonByIndex(partyMenu->args->party, partyMenu->partyMonIndex), MON_DATA_HP, NULL);
+    // Hook: Speed up HP restore here
+    // This currently animates 1 HP per frame
+    if (hp != partyMenu->monsDrawState[partyMenu->partyMonIndex].hp) {
+        ++partyMenu->monsDrawState[partyMenu->partyMonIndex].hp;
+    }
+    PartyMenu_ClearMonHpTextWindow(partyMenu, partyMenu->partyMonIndex);
+    FillWindowPixelBuffer(&partyMenu->windows[PARTY_MENU_WINDOW_ID_MON1_HPBAR + partyMenu->partyMonIndex * 5], 0);
+    PartyMenu_PrintMonCurHpOnWindow(partyMenu, partyMenu->partyMonIndex);
+    PartyMenu_DrawMonHpBarOnWindow(partyMenu, partyMenu->partyMonIndex);
+    if (hp == partyMenu->monsDrawState[partyMenu->partyMonIndex].hp) {
+        PartyMenu_PrintMessageOnWindow34(partyMenu, -1, TRUE);
+        partyMenu->itemUseCallback = PartyMenu_ItemUseFunc_WaitTextPrinterThenExit;
+    }
+    return PARTY_MENU_STATE_ITEM_USE_CB;
+}
+
+BOOL ItemId_IsReviveAll(u16 itemId) {
+    return !!GetItemAttr(itemId, ITEMATTR_REVIVE_ALL, HEAP_ID_PARTY_MENU);
+}
+
+static u8 PartyMenu_SacredAshGetNextMonId(PartyMenuStruct *partyMenu, u8 partySlot) {
+    if (partySlot >= PARTY_SIZE) {
+        return 0xFF;
```

Comment:
PARTY_SLOT_NONE

Reply:
not creating a whole constant just for this case